===================================================================
--- src/aaa_template_gui_create.h	(revision 0)
+++ src/aaa_template_gui_create.h	(revision 0)
@@ -0,0 +1,11 @@
+// template creation gui
+
+#ifndef TEMPLATE_GUI_CREATE
+#define TEMPLATE_GUI_CREATE
+
+#include "aaa_template_vehicle.h"
+#include "aaa_template_gui_create_virtualtrain.h"
+
+void ShowTemplateCreateWindow(TemplateVehicle*, bool*, bool*, int);
+
+#endif
