===================================================================
--- projects/openttd_vs90.vcproj	(revision 24056)
+++ projects/openttd_vs90.vcproj	(working copy)
@@ -432,6 +432,54 @@
 			Name="Source Files"
 			>
 			<File
+				RelativePath=".\..\src\aaa_template_gui_main.cpp"
+				>
+			</File>
+			<File
+				RelativePath=".\..\src\aaa_template_gui_create.cpp"
+				>
+			</File>
+			<File
+				RelativePath=".\..\src\aaa_template_gui_create_virtualtrain.cpp"
+				>
+			</File>
+			<File
+				RelativePath=".\..\src\aaa_template_gui_replaceall.cpp"
+				>
+			</File>
+			<File
+				RelativePath=".\..\src\aaa_template_vehicle.cpp"
+				>
+			</File>
+			<File
+				RelativePath=".\..\src\aaa_template_vehicle_func.cpp"
+				>
+			</File>
+			<File
+				RelativePath=".\..\src\aaa_template_gui_main.h"
+				>
+			</File>
+			<File
+				RelativePath=".\..\src\aaa_template_gui_create.h"
+				>
+			</File>
+			<File
+				RelativePath=".\..\src\aaa_template_gui_create_virtualtrain.h"
+				>
+			</File>
+			<File
+				RelativePath=".\..\src\aaa_template_gui_replaceall.h"
+				>
+			</File>
+			<File
+				RelativePath=".\..\src\aaa_template_vehicle.h"
+				>
+			</File>
+			<File
+				RelativePath=".\..\src\aaa_template_vehicle_func.h"
+				>
+			</File>
+			<File
 				RelativePath=".\..\src\airport.cpp"
 				>
 			</File>
@@ -2557,6 +2605,18 @@
 			</File>
 		</Filter>
 		<Filter
+			Name="MYGUI SL HANDLERS"
+			>
+			<File
+				RelativePath=".\..\src\saveload\aaa_template_replacement_sl.cpp"
+				>
+			</File>
+			<File
+				RelativePath=".\..\src\saveload\aaa_template_veh_sl.cpp"
+				>
+			</File>
+		</Filter>
+		<Filter
 			Name="Tables"
 			>
 			<File
