===================================================================
--- projects/openttd_vs100.vcxproj	(revision 24056)
+++ projects/openttd_vs100.vcxproj	(working copy)
@@ -291,6 +291,18 @@
     </Link>
   </ItemDefinitionGroup>
   <ItemGroup>
+    <ClCompile Include="..\src\aaa_template_gui_main.cpp" />
+    <ClCompile Include="..\src\aaa_template_gui_create.cpp" />
+    <ClCompile Include="..\src\aaa_template_gui_create_virtualtrain.cpp" />
+    <ClCompile Include="..\src\aaa_template_gui_replaceall.cpp" />
+    <ClCompile Include="..\src\aaa_template_vehicle.cpp" />
+    <ClCompile Include="..\src\aaa_template_vehicle_func.cpp" />
+    <ClInclude Include="..\src\aaa_template_gui_main.h" />
+    <ClInclude Include="..\src\aaa_template_gui_create.h" />
+    <ClInclude Include="..\src\aaa_template_gui_create_virtualtrain.h" />
+    <ClInclude Include="..\src\aaa_template_gui_replaceall.h" />
+    <ClInclude Include="..\src\aaa_template_vehicle.h" />
+    <ClInclude Include="..\src\aaa_template_vehicle_func.h" />
     <ClCompile Include="..\src\airport.cpp" />
     <ClCompile Include="..\src\animated_tile.cpp" />
     <ClCompile Include="..\src\articulated_vehicles.cpp" />
@@ -816,6 +828,8 @@
     <ClCompile Include="..\src\saveload\town_sl.cpp" />
     <ClCompile Include="..\src\saveload\vehicle_sl.cpp" />
     <ClCompile Include="..\src\saveload\waypoint_sl.cpp" />
+    <ClCompile Include="..\src\saveload\aaa_template_replacement_sl.cpp" />
+    <ClCompile Include="..\src\saveload\aaa_template_veh_sl.cpp" />
     <ClInclude Include="..\src\table\airport_defaults.h" />
     <ClInclude Include="..\src\table\airport_movement.h" />
     <ClInclude Include="..\src\table\airporttile_ids.h" />
