===================================================================
--- source.list	(revision 24056)
+++ source.list	(working copy)
@@ -1,4 +1,17 @@
 # Source Files
+aaa_template_gui_main.cpp
+aaa_template_gui_create.cpp
+aaa_template_gui_create_virtualtrain.cpp
+aaa_template_gui_replaceall.cpp
+aaa_template_vehicle.cpp
+aaa_template_vehicle_func.cpp
+aaa_template_gui_main.h
+aaa_template_gui_create.h
+aaa_template_gui_create_virtualtrain.h
+aaa_template_gui_replaceall.h
+aaa_template_vehicle.h
+aaa_template_vehicle_func.h
+
 airport.cpp
 animated_tile.cpp
 articulated_vehicles.cpp
@@ -574,6 +587,9 @@
 saveload/town_sl.cpp
 saveload/vehicle_sl.cpp
 saveload/waypoint_sl.cpp
+# MYGUI SL HANDLERS
+saveload/aaa_template_replacement_sl.cpp
+saveload/aaa_template_veh_sl.cpp
 
 # Tables
 table/airport_defaults.h
